### PR TITLE
SYNCT-251: Added service and DAO for the ParentObjectHashcode

### PR DESCRIPTION
**Pull request by areklalo**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/sync2/api/model/ParentObjectHashcode.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package org.openmrs.module.sync2.api.model;
+
+import org.openmrs.BaseOpenmrsData;
+
+import java.util.Objects;
+
+public class ParentObjectHashcode extends BaseOpenmrsData {
+
+	private Integer id;
```
Comment (rstencel) [inline api/src/main/java/org/openmrs/module/sync2/api/model/ParentObjectHashcode.java]: Is this id of hashcode entity itself or of local object? In order to search through hashcodes stored on local instance we were supposed to save local objects ids. Did you solve it some other way?

Comment (areklalo) [reply api/src/main/java/org/openmrs/module/sync2/api/model/ParentObjectHashcode.java]: This is the database id. In order to search hashcode we use the UUID of the object.

File: api/src/main/resources/liquibase.xml [hunk 2]
```diff
@@ -62,6 +62,40 @@
...
+    <changeSet id="sync2-2018-11-13-20:00" author="Arkadiusz Lalo">
+        <preConditions>
+            <not>
+                <tableExists tableName="sync_parent_object_hashcode" />
+            </not>
+        </preConditions>
+        <comment>
+            Creating the table for hashes of synchronized objects.
+        </comment>
+        <createTable tableName="sync_parent_object_hashcode">
```
Comment (rstencel) [inline api/src/main/resources/liquibase.xml]: Is "hashcode" field missing?

File: api/src/main/resources/ParentObjectHashcode.hbm.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE hibernate-mapping PUBLIC "-//Hibernate/Hibernate Mapping DTD 3.0//EN" "http://hibernate.sourceforge.net/hibernate-mapping-3.0.dtd">
+<hibernate-mapping package="org.openmrs.module.sync2.api.model">
+    <class name="ParentObjectHashcode" table="sync_parent_object_hashcode">
+        <id column="parent_object_hashcode_id" name="id" unsaved-value="0">
+            <generator class="native">
+                <param name="sequence">parent_object_hashcode_id_seq</param>
+            </generator>
+        </id>
+        <property column="object_uuid" name="objectUuid"/>
```
[the remark this answers is not in the document]
Comment (areklalo) [reply api/src/main/resources/ParentObjectHashcode.hbm.xml]: You are right, corrected.

File: api/src/test/java/org/openmrs/module/sync2/api/service/impl/ParentObjectHashcodeServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+	@Test
+	public void getByUuid_shouldFetchCorrectObject() {
+		ParentObjectHashcode expected = ParentObjectHashcodeMother.createInstance(OBJECT_ID);
+		ParentObjectHashcode actual = parentObjectHashcodeService.getByUuid(ParentObjectHashcodeMother.UUID);
+		Assert.assertEquals(expected, actual);
+	}
+
+	@Test
+	public void getByObjetUuid_shouldFetchCorrectObject() {
```
[the remark this answers is not in the document]
Comment (areklalo) [reply api/src/test/java/org/openmrs/module/sync2/api/service/impl/ParentObjectHashcodeServiceImplTest.java]: Nice, thanks.

File: api/src/main/resources/liquibase.xml (hunk 2 above)
Comment (areklalo) [reply api/src/main/resources/liquibase.xml]: Same as above.

File: api/src/main/resources/moduleApplicationContext.xml [hunk 5]
```diff
@@ -71,10 +70,19 @@
     <bean parent="serviceContext">
         <property name="moduleService">
             <list>
-                <value>org.openmrs.module.sync2.api.service.MergeConflictService</value>
-                <ref bean="sync2.mergeConflictService" />
```
[the remark this answers is not in the document]
Comment (areklalo) [reply api/src/main/resources/moduleApplicationContext.xml]: On the bottom of this file you could find this service.

**Pull request merged**